plugins/templates/get.js

```diff
@@ -5,27 +5,59 @@ const joi = require('joi');
 const schema = require('screwdriver-data-schema');
 const getSchema = schema.models.template.get;
 const baseSchema = schema.models.template.base;
+const versionRegex = schema.config.regex.VERSION;
+const versionSchema = joi.reach(baseSchema, 'version');
 
 module.exports = () => ({
     method: 'GET',
-    path: '/templates/{name}/{version}',
+    path: '/templates/{name}/{versionOrTag}',
     config: {
-        description: 'Get a single template',
+        description: 'Get a single template given template name and version or tag',
         notes: 'Returns a template record',
         tags: ['api', 'templates'],
         handler: (request, reply) => {
-            const factory = request.server.app.templateFactory;
-
-            return factory.getTemplate({
-                name: request.params.name,
-                version: request.params.version
-            }).then((template) => {
-                if (!template) {
-                    throw boom.notFound('Template does not exist');
+            const versionOrTag = request.params.versionOrTag;
+            const templateFactory = request.server.app.templateFactory;
+
+            // check if version or tag
+            const isVersion = versionOrTag.match(versionRegex);
```

Comment:
> Do we have any formal requirements for what a tag should/shouldn't be?

I'm unsure. 

Because of the way that the regex is formatted, the user cannot use a tag that is only composed of numbers. So this means that a user cannot use a simplified date stamp (e.g., `20170731`) or time (e.g., a unix timestamp like `1500491037`).